test/unit/math/mix/mat/prob/std_binormal_lcdf_test.cpp

```diff
@@ -0,0 +1,1245 @@
+                       const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec,
+                       Eigen::Matrix<T, Eigen::Dynamic, 1>& y, T& rho) {
+  y(0) = inp_vec(0);
+  y(1) = inp_vec(1);
+  rho = T(inp_vec(2));
+}
+
+// Row Vector, real
+template <typename T>
+void to_function_input(int N_y,
+                       const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec,
+                       Eigen::Matrix<T, 1, Eigen::Dynamic>& y, T& rho) {
+  y(0) = inp_vec(0);
+  y(1) = inp_vec(1);
+  rho = T(inp_vec(2));
+}
+
+// vector<Vector>, real
+template <typename T>
+void to_function_input(int N_y,
+                       const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec,
+                       vector<Eigen::Matrix<T, Eigen::Dynamic, 1>>& y, T& rho) {
+  int cntr = 0;
+  for (int i = 0; i < N_y; ++i) {
+    Eigen::Matrix<T, Eigen::Dynamic, 1> el(2);
+    el(0) = inp_vec(cntr);
+    ++cntr;
+    el(1) = inp_vec(cntr);
+    ++cntr;
+    y.push_back(el);
+  }
+  rho = inp_vec(N_y * 2);
+}
+
+// vector<Row Vector>, real
+template <typename T>
+void to_function_input(int N_y,
+                       const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec,
+                       vector<Eigen::Matrix<T, 1, Eigen::Dynamic>>& y, T& rho) {
+  int cntr = 0;
+  for (int i = 0; i < N_y; ++i) {
+    Eigen::Matrix<T, Eigen::Dynamic, 1> el(2);
+    el(0) = inp_vec(cntr);
+    ++cntr;
+    el(1) = inp_vec(cntr);
+    ++cntr;
+    y.push_back(el);
+  }
+  rho = inp_vec(N_y * 2);
+}
+
+// vector<Vector>, vector
+template <typename T>
+void to_function_input(int N_y,
+                       const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec,
+                       vector<Eigen::Matrix<T, Eigen::Dynamic, 1>>& y,
+                       vector<T>& rho) {
+  int cntr = 0;
+  for (int i = 0; i < N_y; ++i) {
+    Eigen::Matrix<T, Eigen::Dynamic, 1> el(2);
+    el(0) = inp_vec(cntr);
+    ++cntr;
+    el(1) = inp_vec(cntr);
+    ++cntr;
+    y.push_back(el);
+  }
+  for (int i = cntr; i < inp_vec.size(); ++i)
+    rho.push_back(T(inp_vec(i)));
+}
+
+// vector<Row Vector>, vector
+template <typename T>
+void to_function_input(int N_y,
+                       const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec,
+                       vector<Eigen::Matrix<T, 1, Eigen::Dynamic>>& y,
+                       vector<T>& rho) {
+  int cntr = 0;
+  for (int i = 0; i < N_y; ++i) {
+    Eigen::Matrix<T, Eigen::Dynamic, 1> el(2);
+    el(0) = inp_vec(cntr);
+    ++cntr;
+    el(1) = inp_vec(cntr);
+    ++cntr;
+    y.push_back(el);
+  }
+  for (int i = cntr; i < inp_vec.size(); ++i)
+    rho.push_back(T(inp_vec(i)));
+}
+
+// vector<Vector>, Vector
+template <typename T>
+void to_function_input(int N_y,
+                       const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec,
+                       vector<Eigen::Matrix<T, Eigen::Dynamic, 1>>& y,
+                       Eigen::Matrix<T, Eigen::Dynamic, 1>& rho) {
+  int cntr = 0;
+  for (int i = 0; i < N_y; ++i) {
```

Comment:
More than one loop variable can be used:

```
for (int i = 0, cntr = 0; ...)
```

then it's clear it's just used in the loop.